src/dmd/traits.d

```diff
@@ -1332,7 +1332,17 @@ extern (C++) Expression semanticTraits(TraitsExp e, Scope* sc)
             return 0;
         }
 
-        ScopeDsymbol._foreach(sc, sds.members, &pushIdentsDg);
+        __gshared int counter;
+        if (counter > 1000)
+        {
+            e.error("Maximal recursion depth hit");
```

Comment:
I recommend "recursion limit exceeded" to remain consistent with the wording used at https://github.com/dlang/dmd/blob/master/src/dmd/dinterpret.d#L927

Reply:
it can't hurt to make it explicit.